gcloud/pubsub/client.py

```diff
@@ -43,6 +46,28 @@ class Client(JSONClient):
     """
 
     _connection_class = Connection
+    _publisher_api = _subscriber_api = _iam_policy_api = None
+
+    @property
+    def publisher_api(self):
+        """Helper for publisher-related API calls."""
+        if self._publisher_api is None:
+            self._publisher_api = _PublisherAPI(self.connection)
```

Comment:
No need to change, just noting that I remain "not a fan" of the lazy-loading approach for cases like this, since the branch occurs on every access.